internal/secrets/gcpkms/kms_test.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcpkms
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cloud/internal/testing/setup"
+	"google.golang.org/api/option"
+
+	cloudkms "cloud.google.com/go/kms/apiv1"
+	"github.com/google/go-cloud/internal/secrets/driver"
+	"github.com/google/go-cloud/internal/secrets/drivertest"
+)
+
+const (
+	// These constants capture values that were used during the last --record.
+	// If you want to use --record mode,
+	// 1. Update projectID to your GCP project name (not number!)
+	// 2. Enable the Cloud KMS API.
+	// 3. Create a key ring and a key, change their name below accordingly.
+	projectID = "pledged-solved-practically"
+	location  = "global"
+	keyRing   = "test"
+	keyID     = "password"
+)
+
+type harness struct {
+	client *cloudkms.KeyManagementClient
+	close  func()
+}
+
+func (h *harness) MakeDriver(ctx context.Context) (driver.Crypter, error) {
+	return &Crypter{
+		keyInfo: &KeyInfo{
+			ProjectID: projectID,
+			Location:  location,
+			KeyRing:   keyRing,
+			KeyID:     keyID,
+		},
+		client: h.client,
+	}, nil
+}
+
+func (h *harness) Close() {
+	h.close()
+}
+
+func newHarness(ctx context.Context, t *testing.T) (drivertest.Harness, error) {
+	conn, done := setup.NewGCPgRPCConn(ctx, t, endPoint)
+	client, err := cloudkms.NewKeyManagementClient(ctx, option.WithGRPCConn(conn))
+	if err != nil {
+		return nil, err
+	}
+	return &harness{
+		client: client,
+		close: func() {
+			_ = client.Close()
```

Comment:
Why does `client.Close()` get assigned to anything here? (I am assuming there's a reason it needs to be, just curious what it is.)